buildconfig/Jenkins/buildscript

```diff
@@ -256,69 +260,101 @@ fi
+        PACKAGINGVARS="${PACKAGINGVARS} -DCPACK_PACKAGE_SUFFIX="
     else
-      PACKAGE_SUFFIX="unstable"
+        # Use different suffix for linux builds if parameter is not defined
+        if [[ -n "${PACKAGE_SUFFIX}" ]]; then
+            echo "Using PACKAGE_SUFFIX=${PACKAGE_SUFFIX} from job parameter"
+            elif [[ ${JOB_NAME} == *release* ]]; then
+            PACKAGE_SUFFIX=""
+            elif [[ ${JOB_NAME} == *master* ]]; then
+            PACKAGE_SUFFIX="nightly"
+            elif [[ ${JOB_NAME} == *pvnext* ]]; then
+            PACKAGE_SUFFIX="mantidunstable-pvnext"
+        else
+            PACKAGE_SUFFIX="unstable"
+        fi
+        
+        if [[ ${PY3_BUILD} == true ]]; then
+            # Add '-python3' to package name and install path
+            PACKAGE_SUFFIX=${PACKAGE_SUFFIX}-python3
+        fi
+        
+        if [[ ${JOB_NAME} == *release* ]] && [[ ${PY2_BUILD} == true ]] && [[ -z "${PACKAGE_SUFFIX}" ]]; then
+            # Traditional install path for python 2 release build
+            PACKAGINGVARS="${PACKAGINGVARS} -DCMAKE_INSTALL_PREFIX=/opt/Mantid -DCPACK_PACKAGE_SUFFIX="
+        else
+            # everything else uses lower-case values
+            PACKAGINGVARS="${PACKAGINGVARS} -DCMAKE_INSTALL_PREFIX=/opt/mantid${PACKAGE_SUFFIX} -DCPACK_PACKAGE_SUFFIX=${PACKAGE_SUFFIX}"
+        fi
+        
+        if [[ ${ON_RHEL7} == true ]]; then
+            if [[ -n "${RELEASE_NUMBER}" ]]; then
+                RELEASE_NUMBER="1"
+            fi
+            PACKAGINGVARS="${PACKAGINGVARS} -DCPACK_RPM_PACKAGE_RELEASE=${RELEASE_NUMBER}"
+        fi
     fi
+fi
 
-    if [[ ${PY3_BUILD} == true ]]; then
-      # Add '-python3' to package name and install path
-      PACKAGE_SUFFIX=${PACKAGE_SUFFIX}-python3
-    fi
+###############################################################################
+# Figure out if were doing a sanitizer build and setup any steps we need
+###############################################################################
+ASAN_OPTIONS=''
+LSAN_OPTIONS=''
 
-    if [[ ${JOB_NAME} == *release* ]] && [[ ${PY2_BUILD} == true ]] && [[ -z "${PACKAGE_SUFFIX}" ]]; then
-      # Traditional install path for python 2 release build
-      PACKAGINGVARS="${PACKAGINGVARS} -DCMAKE_INSTALL_PREFIX=/opt/Mantid -DCPACK_PACKAGE_SUFFIX="
-    else
-      # everything else uses lower-case values
-      PACKAGINGVARS="${PACKAGINGVARS} -DCMAKE_INSTALL_PREFIX=/opt/mantid${PACKAGE_SUFFIX} -DCPACK_PACKAGE_SUFFIX=${PACKAGE_SUFFIX}"
-    fi
+SANITIZER_FLAGS=''
 
-    if [[ ${ON_RHEL7} == true ]]; then
-      if [[ -n "${RELEASE_NUMBER}" ]]; then
-        RELEASE_NUMBER="1"
-      fi
-      PACKAGINGVARS="${PACKAGINGVARS} -DCPACK_RPM_PACKAGE_RELEASE=${RELEASE_NUMBER}"
-    fi
-  fi
+SUPPRESSIONS_DIR="${WORKSPACE}/buildconfig/Sanitizer"
+
+if [[ ${JOB_NAME} == *address* ]]; then
+    SANITIZER_FLAGS="-DUSE_SANITIZER=Address"
+    
+    # At a future date we could add an LD_PRELOAD export here, see docs for info
```

Comment:
My understanding is that gcc 8 is required to get the `LD_PRELOAD` of `asan` to work properly? It would be good to not require that for the C++ tests if possible so that it's easier for developers to run this. Maybe we disable the python tests if we have sanitizers enabled and gcc < 8? We could do this across the board quite easily in [PyUnitTest.cmake](https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/PyUnitTest.cmake#L14) and print a warning that they are disabled.

For gcc >= 8 we could set the `LD_PRELOAD` option inside [`mantidpython`](https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/LinuxPackageScripts.cmake#L164) so that the python tests will grab it automatically.

Reply:
The only difficulty is generating the path to libasan

We need to know which version of GCC was used to select the right asan version...etc. Something like : `ldd path/to/mantid | grep libasan | sed "s/^[[:space:]]//" | cut -d' ' -f1` would work, but it requires the build to complete to get the path at configure time

I'm tempted to leave this to the future since having ASAN on C++ tests currently is a good start

Reply:
@DavidFair Agreed. Starting with the C++ tests would be more than adequate.